### ISSUE TITLE
Unable to extract the compressed linux file system - Debian Live 11

### ISSUE THREAD

**Issue by acdcjavier** [PJ-Singh-001/Cubic#339]
**Describe the bug**
When I select the iso to customize, appears the error: Unable to extract the compressed linux file system

**To Reproduce**

Download Debian 11 live ISO:
https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/archive/11.11.0-live+nonfree/amd64/iso-hybrid/
and choose:
debian-live-11.11.0-amd64-xfce+nonfree.iso

**Expected behavior**

ISO must be uncompressed as expected

**OS Information (please complete the following information):**
 - OS Version Debian Live 11.11.0

**Cubic Information (please complete the following information):**
 - Cubic Version: 2024.09-88-release~202409021406~ubuntu22.04.1 all
 - Download Link:  [The download link for the ISO you are customizing](https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/archive/11.11.0-live+nonfree/amd64/iso-hybrid/)

**Cubic Log:**

[log cubic.txt](https://github.com/user-attachments/files/16852182/log.cubic.txt)

**Comment by PJ-Singh-001** [comment]
I just tested debian-live-12.7.0-amd64-gnome.iso, and I was able to generate an ISO using Cubic.

I've downloaded debian-live-11.11.0-amd64-xfce+nonfree.iso that you linked to above, and I'll look at it to see what is going on.

**Comment by PJ-Singh-001** [comment]
Cubic can only customize the "Live" ISOs (which is the _variant_ you had indicated you are customizing).

Double check you that had downloaded [`debian-live-11.11.0-amd64-xfce+nonfree.iso`](https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/archive/11.11.0-live+nonfree/amd64/iso-hybrid/debian-live-11.11.0-amd64-xfce+nonfree.iso).

After confirming that you had downloaded the correct `*.iso` file, run Cubic using `cubic --verbose`.
(You can also run Cubic using `cubic --verbose --log`, and a log will be output to the terminal _and_ saved to your project directory).
As soon as you experience the error, cut and paste the log here.

Here are screenshots of my successful test customizing `debian-live-11.11.0-amd64-xfce+nonfree.iso` in Cubic 2024.09.88 on in Ubuntu 24.04...

![Screenshot from 2024-09-03 14-35-27](https://github.com/user-attachments/assets/5304c789-edb0-40f9-b9ec-583665b897f2)

![Screenshot from 2024-09-03 14-38-43](https://github.com/user-attachments/assets/80752421-4ee6-4ca5-82bf-93c6668602d0)

**Comment by acdcjavier** [comment]
Thank I will re check again
The difference is that I'm using Debian 11 as SO to use Cubic and you are using Ubuntu 24.04

**Comment by PJ-Singh-001** [comment]
> Thank I will re check again The difference is that I'm using Debian 11 as SO to use Cubic and you are using Ubuntu 24.04

The log will show us if there is an error that occurs only when running Cubic on Debian 11.

**Comment by acdcjavier** [comment]
> > Thank I will re check again The difference is that I'm using Debian 11 as SO to use Cubic and you are using Ubuntu 24.04
> 
> The log will show us if there is an error that occurs only when running Cubic on Debian 11.

I have reproduced again the issue:

Attach log
[cubic.20240904102505.log](https://github.com/user-attachments/files/16872229/cubic.20240904102505.log)

**Comment by PJ-Singh-001** [comment]
Thanks. I see an error in your logs.

Let's check what versions of `squashfs-tools` and Python you have.
What is the output of these commands?

    dpkg -l squashfs-tools

    python3 --version

Let's try to extract the squash file system manually, and see if you get an error.
Please paste the output of the following commands here.

    # Mount the ISO.
    sudo mount --options loop,uid=1000,gid=1000 /home/mbstbtester11/Descargas/debian-live-11.11.0-amd64-xfce+nonfree.iso /home/mbstbtester11/ISO/source-disk

    # Extract the squashfs.
    sudo unsquashfs -force -no-exit-code -dest /home/mbstbtester11/ISO/custom-root /home/mbstbtester11/ISO/source-disk/live/filesystem.squashfs

When you are done, you can unmount the ISO by executing the following:

    sudo umount /home/mbstbtester11/ISO/source-disk

**Comment by PJ-Singh-001** [comment]
Update:

Thank you for the log; it was helpful in identifying the problem.
I installed Debian 11.11, and I was able to recreate the issue as well.

I'll post a solution tomorrow.

**Comment by PJ-Singh-001** [comment]
I've updated the [installation instructions for Cubic in Debian Bullseye](https://github.com/PJ-Singh-001/Cubic/wiki/Install-Cubic#-debian-and-derivatives).

First, remove the version of Cubic you have installed.

    sudo apt autoremove --purge cubic
    sudo apt autoremove --purge dpkg

To install and run Cubic in Debian 11, temporarily use the `bookworm` repositories to install newer versions of the `dpkg` and `squashfs-tools` packages. _Be careful with this hack, as this may break your system._

Install `dpkg` version 1.21.22 from Debian `bookworm` (instead of version 1.20.13 available in the `bullseye` repositories), and install `squashfs-tools` version 1:4.5.1 from Debian `bookworm` (instead of version 1:4.4 available in the `bullseye` repositories).

    cd /etc/apt
    sudo cp sources.list sources.list.original
    sudo sed -i "s|bullseye|bookworm|g" sources.list
    sudo apt update

    sudo apt install --no-install-recommends dpkg
    sudo apt install --no-install-recommends squashfs-tools

    sudo mv sources.list.original sources.list
    sudo apt update

Install the latest version of Cubic (version 2024.09.88) from the _noble_ packages repository. (Ubuntu 24.04.1 LTS _Noble_ Numbat is the most recent long term support version of Ubuntu).

    echo "deb https://ppa.launchpadcontent.net/cubic-wizard/release/ubuntu/ noble main" | sudo tee /etc/apt/sources.list.d/cubic-wizard-release.list
    curl -S "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x081525e2b4f1283b" | sudo gpg --batch --yes --dearmor --output /etc/apt/trusted.gpg.d/cubic-wizard-ubuntu-release.gpg
    sudo apt update
    sudo apt install --no-install-recommends cubic

The `--no-install-recommends` option prevents superfluous packages from being installed.

**Comment by acdcjavier** [comment]
> I've updated the [installation instructions for Cubic in Debian Bullseye](https://github.com/PJ-Singh-001/Cubic/wiki/Install-Cubic#-debian-and-derivatives).
> 
> First, remove the version of Cubic you have installed.
> 
> ```
> sudo apt autoremove --purge cubic
> sudo apt autoremove --purge dpkg
> ```
> 
> To install and run Cubic in Debian 11, temporarily use the `bookworm` repositories to install newer versions of the `dpkg` and `squashfs-tools` packages. _Be careful with this hack, as this may break your system._
> 
> Install `dpkg` version 1.21.22 from Debian `bookworm` (instead of version 1.20.13 available in the `bullseye` repositories), and install `squashfs-tools` version 1:4.5.1 from Debian `bookworm` (instead of version 1:4.4 available in the `bullseye` repositories).
> 
> ```
> cd /etc/apt
> sudo cp sources.list sources.list.original
> sudo sed -i "s|bullseye|bookworm|g" sources.list
> sudo apt update
> 
> sudo apt install --no-install-recommends dpkg
> sudo apt install --no-install-recommends squashfs-tools
> 
> sudo mv sources.list.original sources.list
> sudo apt update
> ```
> 
> Install the latest version of Cubic (version 2024.09.88) from the _noble_ packages repository. (Ubuntu 24.04.1 LTS _Noble_ Numbat is the most recent long term support version of Ubuntu).
> 
> ```
> echo "deb https://ppa.launchpadcontent.net/cubic-wizard/release/ubuntu/ noble main" | sudo tee /etc/apt/sources.list.d/cubic-wizard-release.list
> curl -S "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x081525e2b4f1283b" | sudo gpg --batch --yes --dearmor --output /etc/apt/trusted.gpg.d/cubic-wizard-ubuntu-release.gpg
> sudo apt update
> sudo apt install --no-install-recommends cubic
> ```
> 
> The `--no-install-recommends` option prevents superfluous packages from being installed.

Thanks! Now worked to uncompress the ISO
However, now when I've finished the customization and select the kernel I got this issue during generating the new ISO:

"Error: Unable to update the vmlinuz boot file"

Attach log:
[cubic.20240906094619.log](https://github.com/user-attachments/files/16908911/cubic.20240906094619.log)

**Comment by PJ-Singh-001** [comment]
OK.
Please open a separate issue for the "Unable to update the vmlinuz boot file" problem.

**Comment by PJ-Singh-001** [comment]
UPDATE:

With Cubic 2024.09.89, you no longer need to install version 1:4.5.1 of `squashfs-tools` from Debian `bookworm`.
Cubic now works with the previous version of `squashfs-tools` (version 1:4.4), available in the `bullseye` repositories.

I've updated the [installation instructions for Cubic in Debian Bullseye](https://github.com/PJ-Singh-001/Cubic/wiki/Install-Cubic#-debian-and-derivatives) to reflect this change.

(This fix was implemented for issue #341, [Cubic can't extract files from original disk #341](https://github.com/PJ-Singh-001/Cubic/issues/341)).
